### PR TITLE
Deployment Utility

### PR DESCRIPTION
**Pull request by barakman**
Extract a deployment utility from the test_deployment script, and use it separately from that script and from a hard-hat test file.

### CODE REVIEW

**Comment by barakman** [comment]
Phase 1:
- The hard-hat test executes the Deployment utility using ethers.js functionality.
- The deployment script executes the Deployment utility using web3.js functionality.

In phase 2, we may want to consider porting the deployment script from web3.js to ethers.js.

**Comment by sshmaxime** [comment]
Nice ! Might as well do the phase 2 in this PR no ? @barakman

**Review by yudilevi — APPROVED** [review]

File: test/Deployment.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
+const Contracts = require('./helpers/Contracts');
+const runDeployment = require('./helpers/runDeployment');
+
+const config =
+{
+    "reserves": [
```
Comment (lbeder) [inline test/Deployment.js:6]: These shouldn't be quoted, since it's not a JSON file

File: test/Deployment.js [hunk 2]
```diff
@@ -0,0 +1,101 @@
+const Contracts = require('./helpers/Contracts');
+const runDeployment = require('./helpers/runDeployment');
+
+const config =
+{
```
Comment (lbeder) [inline test/Deployment.js:5]: The bracket should be in the same line. Did you format these files?

File: utils/test_deployment.js [hunk 3]
```diff
@@ -2,19 +2,16 @@ const fs = require('fs');
 const path = require('path');
 const Web3 = require('web3');
 
+const runDeployment = require('../test/helpers/runDeployment');
```
Comment (lbeder) [inline utils/test_deployment.js:5]: 👍 👍 👍

**Review by lbeder — CHANGES_REQUESTED** [review]
Nice job! Please see some minor comment above

File: test/Deployment.js (hunk 1 above)
Comment (barakman) [reply test/Deployment.js:6]: ah yeah, copy-paste (from the README file) mistake

File: test/Deployment.js (hunk 2 above)
Comment (barakman) [reply test/Deployment.js:5]: ah yeah, copy-paste (from the README file) mistake

(yes, I actually did format these files afaik)

**Comment by barakman** [comment]
> Nice ! Might as well do the phase 2 in this PR no ? @barakman

I am not yet familiar with signing transactions using ethers.js.
I am guessing that it might take a while to implement everything in a relatively robust manner (as robust as it is today, or more).

File: test/Deployment.js (hunk 1 above)
Comment (barakman) [reply test/Deployment.js]: fixed

File: test/Deployment.js (hunk 2 above)
Comment (barakman) [reply test/Deployment.js]: fixed

**Review by lbeder — APPROVED** [review]

**Pull request merged**